C++ API parity: LogSigmoid

pbelevich opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #27309 C++ API parity: MultiheadAttention
* #27382 C++ API parity: Linear
* **#27060 C++ API parity: LogSigmoid**

Differential Revision: [D17682404](https://our.internmc.facebook.com/intern/diff/D17682404)

pbelevich committed:
C++ API parity: LogSigmoid

[ghstack-poisoned]

yf225 commented on torch/csrc/api/include/torch/nn/modules/activation.h:
```diff
@@ -101,5 +101,27 @@ class TORCH_API LeakyReLUImpl : public torch::nn::Cloneable<LeakyReLUImpl> {
+  void pretty_print(std::ostream& stream) const override;
+
+  /// The options with which this `Module` was constructed.
+  LeakyReLUOptions options;
```
We should remove this

yf225 commented on torch/csrc/api/include/torch/nn/functional/activation.h:
```diff
@@ -37,6 +37,10 @@ inline Tensor leaky_relu(const Tensor& input, const LeakyReLUOptions& options) {
   }
 }
 
+inline Tensor log_sigmoid(const Tensor& input) {
```
I think the Python functional is called `logsigmoid`, we probably should name it the same here.

pbelevich committed:
Update on "C++ API parity: LogSigmoid"




[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: LogSigmoid"




[ghstack-poisoned]

yf225 reviewed (approved):
Thanks @pbelevich !

pbelevich committed:
Update on "C++ API parity: LogSigmoid"




[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: LogSigmoid"


Differential Revision: [D17682404](https://our.internmc.facebook.com/intern/diff/D17682404)

[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: LogSigmoid"


Differential Revision: [D17682404](https://our.internmc.facebook.com/intern/diff/D17682404)

[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: LogSigmoid"


Differential Revision: [D17682404](https://our.internmc.facebook.com/intern/diff/D17682404)

[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: LogSigmoid"


Differential Revision: [D17682404](https://our.internmc.facebook.com/intern/diff/D17682404)

[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: LogSigmoid"


Differential Revision: [D17682404](https://our.internmc.facebook.com/intern/diff/D17682404)

[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: LogSigmoid"


Differential Revision: [D17682404](https://our.internmc.facebook.com/intern/diff/D17682404)

[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: LogSigmoid"


Differential Revision: [D17682404](https://our.internmc.facebook.com/intern/diff/D17682404)

[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: LogSigmoid"


Differential Revision: [D17682404](https://our.internmc.facebook.com/intern/diff/D17682404)

[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: LogSigmoid"


Differential Revision: [D17682404](https://our.internmc.facebook.com/intern/diff/D17682404)

[ghstack-poisoned]

pbelevich committed:
Update on "C++ API parity: LogSigmoid"


Differential Revision: [D17682404](https://our.internmc.facebook.com/intern/diff/D17682404)

[ghstack-poisoned]
